tests/command.deploy.test.js

```diff
@@ -348,6 +348,56 @@ if (process.env.IS_FORK !== 'true') {
     })
   })
 
+  test.only('should refresh configuration when --build is passed', async (t) => {
+    await withSiteBuilder('site-with-public-folder', async (builder) => {
+      await builder
+        .withContentFile({
+          path: 'public/index.html',
+          content: '<h1>⊂◉‿◉つ</h1>',
+        })
+        .withNetlifyToml({
+          config: {
+            build: { publish: 'public' },
+            functions: { directory: 'functions' },
+            plugins: [{ package: './plugins/mutator' }],
+          },
+        })
+        .withBuildPlugin({
+          name: 'mutator',
+          plugin: {
+            onPreBuild: async ({ netlifyConfig }) => {
+              // eslint-disable-next-line node/global-require
+              const [fs, util] = [require('fs'), require('util')]
```

Comment:
[sand] Would this work at the top-level?

Reply:
This translates to plugin code, so it having it top level in this file will result in missing requires in the plugin file.

Reply:
:+1: